src/OpenLoco/src/Vehicles/OrderManager.cpp

```diff
@@ -404,6 +404,10 @@ namespace OpenLoco::Vehicles::OrderManager
         {
             auto rawOrder = *it;
             auto orderType = rawOrder & 0x7;
+            if (orderType >= std::size(kOrderSizes))
+            {
+                throw std::out_of_range("Order type is greater than the order size table");
```

Comment:
Would be better to use our custom exception type which also adds information from where its thrown. See Core/Exception.hpp.
You have to add a new type for this one.
```
using OutOfRange = Detail::ExceptionBase<struct OutOfRangeTag>;
```

Reply:
I can do that. We do have a couple of existing std::out_of_range calls though, should we change those?

Reply:
Yep, I've must have missed them last time somehow.

Reply:
these don't exist in master - is this something ORCT2 has?

Reply:
oh i see what you mean - i might leave this to another PR